theano/gof/destroyhandler.py

```diff
@@ -1026,3 +1027,23 @@ def orderings(self, fgraph):
                         rval[app] = root_clients
 
         return rval
+
+
+class NoOutputFromInplace(Feature):
+
+    def __init__(self):
+        pass
+
+    def validate(self, fgraph):
+        if not hasattr(fgraph, 'destroyers'):
+            return True
+        for out in list(fgraph.outputs):
+            # Validate that the node that produces the output does not produce
+            # it by modifying something else inplace.
+            node = out.owner
+            op = node.op
```

Comment:
out.owner can be None, if we pass an input of the graph and some compilation of flags to allow this (not allowed by default). So if out.owner is None, just continue the loop.